geode-core/src/main/java/org/apache/geode/internal/admin/remote/ShutdownAllRequest.java

```diff
@@ -293,7 +294,7 @@ public void process(DistributionMessage msg) {
     public Set getResults() {
       logger.debug("{} shutdownAll returning {}", this,
           results/* , new Exception("stack trace") */);
-      return results;
+      return new HashSet(results);
```

Comment:
Shouldn't this method synchronize on "results" for both the logging and for making a copy into the HashSet?

Reply:
yes, and it looks like there's another place that also needs synchronization